cmd/prometheus/main.go

```diff
@@ -254,9 +254,6 @@ func (c *flagConfig) setFeatureListOptions(logger *slog.Logger) error {
 				config.DefaultConfig.GlobalConfig.ScrapeProtocols = config.DefaultProtoFirstScrapeProtocols
 				config.DefaultGlobalConfig.ScrapeProtocols = config.DefaultProtoFirstScrapeProtocols
 				logger.Info("Experimental native histogram support enabled. Changed default scrape_protocols to prefer PrometheusProto format.", "global.scrape_protocols", fmt.Sprintf("%v", config.DefaultGlobalConfig.ScrapeProtocols))
-			case "ooo-native-histograms":
```

Comment:
Back in v2, we did not just remove obsolete feature flags, but had a special warning for them. Quoting from v2.55 code:

```Go
			case "promql-at-modifier", "promql-negative-offset":
				level.Warn(logger).Log("msg", "This option for --enable-feature is now permanently enabled and therefore a no-op.", "option", o)
```

Reply:
Revived flag and added warning: https://github.com/prometheus/prometheus/pull/16207/commits/796015f8e52c757bcdb0f60908244676ada2fae2